src/main/scala/org/broadinstitute/dsde/workbench/sam/service/PolicyEvaluatorService.scala

```diff
@@ -71,23 +75,49 @@ class PolicyEvaluatorService(
       res <- resourceTypes.get(resource.resourceTypeName).traverse {
         rt =>
 
-          // we can only use this approach if the resource type isn't auth-domain constrainable
-          if (!rt.isAuthDomainConstrainable) {
-
-            // compute the list of roles that contain the action requested
-            val roleNamesWithAction = rt.roles.filter(_.actions.contains(action)).map(_.roleName)
+          val roleNamesWithAction = rt.roles.filter(_.actions.contains(action)).map(_.roleName)
+          val hasAction = directMemberHasActionOnResource(resource, roleNamesWithAction, action, userId)
 
+          // if the resource type OR action are NOT auth domain constrainable... just return
+          if (!rt.isAuthDomainConstrainable || !rt.actionPatterns.exists( ap => ap.authDomainConstrainable & ap.matches(action)) ) {
```

Comment:
is there a reason there is only one `&` in `ap.authDomainConstrainable & ap.matches(action)`

Reply:
not a good one ;) . fixed to be &&